PetAdoption-iOS/PetAdoption-iOS/Controllers/PetListingViewController.swift

```diff
@@ -237,15 +261,16 @@ extension PetListingViewController : UICollectionViewDelegate, UICollectionViewD
 {
     func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int
     {
-        return self.petData.count
+        self.noResultsLabel.isHidden = !(self.filteredPetData.count == 0)
```

Comment:
So with endless scroll, there is a collection view footer with a UIActivityIndicatorView in it.  The purpose of that is to give the users an indication that there is something loading.  If the user reaches the very bottom of the collection (2000 results), the activity indicator stops animating.

I noticed a bug whereby if there are no results found, that activity indicator is still animating at the very top.  That's that footer with the activity indicator.  We will probably need to add something somewhere to stop that animation if there are no results.  I'm not sure where exactly to add this, but check out the `viewForSupplementaryElementOfKind at indexPath` function to see where we are currently starting and stopping that animation.

Reply:
On another note, I am currently working on the empty view myself, but what I'll do is leave what you have here and then just add my changes afterwards.

Reply:
Sorry for taking over the empty state issue. I saw it labeled as 'help needed' and assumed it means its ok to pick it up. I'm very new to the whole collaboration way of doing things so thanks for bearing with me :)

About the activity indicator, I think there is a related problem where if not enough animals of a selected type are loaded to fill the screen, a request to download more animals should be triggered, but isn't. This is because pets are loaded into petData, but displayed out of filteredPetData. Not sure what the best way to solve this would be.

Maybe a solution could be to call loadPets() many times while MAX_SEARCH_RESULTS hasn't bean reached AND filteredPetData < minFilteredResultAttempts ?
Any thoughts

Reply:
No worries about the empty state thing.  Typically, if you see someone is assigned to an issue, usually that means they're currently looking into it.  At the very least, reach out to that person to find out if they're actively working on an issue.  Sometimes someone will assign themselves to an issue and then not have any time to actually work on it (happens to me all the time lol).

Let me think about the activity indicator problem a little bit.  For now, I'm ok with leaving this minor bug in there unless we can come up with a solution before we end up pushing the next version.

Reply:
Crap, you know what I just thought about?  I wonder if it would be better for us to actually modify the API call to only return us a list of pets given the animal type.  Going back to the API docs, we can optionally pass a string indicating what type of animals we want returned.  The only issue with this I can see is that we may not be able to filter on multiple animal types.  Well that and a lot of the work you've already put into this will need to be re-done.

One reason we might want to consider this approach is that because we are rate limited to 2000 results max, we are very likely leaving out a lot of results by default.  We could potentially get more results, especially for the less common animals.  This will still leave us with the activity indicator problem because, like you mention, the check for stopping that animation is based solely on the max results (2000).

Sorry for only now thinking about this.  This is probably one of the reasons I had kept this issue out until v1.2.  That and we probably will need to have a discussion about other filtering options we will want to eventually add in.